data/mods/gen2/scripts.js

```diff
@@ -588,7 +588,7 @@ let BattleScripts = {
 		let noburndrop = false;
 
 		// The move is a critical hit. Several things happen here.
-		if (move.crit) {
+		if (isCrit) {
```

Comment:
``this.runEvent('CriticalHit', target, null, move)`` should modify ``isCrit`` so that this check works correctly.